Cherry-pick from the upstream commit: 6dd80faf0

Written and recommended by the upstraem meaintainer: Behdad Esfahbod

diff --git a/src/hb-open-type-private.hh b/src/hb-open-type-private.hh
index 0754ef0..80ad687 100644
--- a/src/hb-open-type-private.hh
+++ b/src/hb-open-type-private.hh
@@ -751,7 +751,7 @@ struct CheckSum : ULONG
 template <typename FixedType=USHORT>
 struct FixedVersion
 {
-  inline uint32_t to_int (void) const { return (major << sizeof(FixedType)) + minor; }
+  inline uint32_t to_int (void) const { return (major << (sizeof(FixedType) * 8)) + minor; }
 
   inline bool sanitize (hb_sanitize_context_t *c) const
   {
